WordPress/src/test/java/org/wordpress/android/ui/reader/repository/usecases/PostLikeUseCaseTest.kt

```diff
@@ -33,58 +50,182 @@ class PostLikeUseCaseTest {
+
+                // Assert
+                assertThat((result)).contains(Success)
+            }
+
+    @Test
+    fun `failure is returned when liking an unliked post`() =
+            testWithBlock {
+                val readerPost = init(isLikedByCurrentUser = false, remoteSucceeds = false)
+
+                // Act
+                val result = useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                assertThat((result)).contains(Failed.RequestFailed)
+            }
+
+    @Test
+    fun `failure is returned when unliking a like post`() =
+            testWithBlock {
+                val readerPost = init(isLikedByCurrentUser = true, remoteSucceeds = false)
+
+                // Act
+                val result = useCase.perform(readerPost, false).toList(mutableListOf())
+
+                // Assert
+                assertThat((result)).contains(Failed.RequestFailed)
+            }
+
+    @Test
+    fun `like local action is triggered for selected reader post`() =
+            testWithBlock {
+                val readerPost = init(isLikedByCurrentUser = false)
+
+                // Act
+                useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                verify(readerPostActionsWrapper).performLikeActionLocal(
+                        anyOrNull(),
+                        anyBoolean(),
+                        anyLong()
+                )
+            }
+
+    @Test
+    fun `like remote action is triggered for selected reader post`() =
+            testWithBlock {
+                val readerPost = init(isLikedByCurrentUser = false)
+
+                // Act
+                useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                verify(readerPostActionsWrapper).performLikeActionRemote(
+                        anyOrNull(),
+                        anyBoolean(),
+                        anyLong(),
+                        anyOrNull()
+                )
+            }
+
+    @Test
+    fun `Post views bumped when asking to like`() =
+            testWithBlock {
+                val readerPost = init(isLikedByCurrentUser = false)
+                // Act
+                useCase.perform(readerPost, true).toList(mutableListOf())
+
+                // Assert
+                verify(readerPostActionsWrapper).bumpPageViewForPost(anyOrNull())
+            }
+
+    @Test
+    fun `Post views NOT bumped when asking to unlike`() =
+
+            testWithBlock {
+                val readerPost = init(isLikedByCurrentUser = true)
+
+                // Act
+                useCase.perform(readerPost, false).toList(mutableListOf())
+
+                // Assert
+                verify(readerPostActionsWrapper, never()).bumpPageViewForPost(anyOrNull())
+            }
+
+    private fun <T> testWithBlock(block: suspend CoroutineScope.() -> T) {
```

Comment:
I believe we can remove this `testWithBlock` and use `test` everywhere, right? The `test` function [runs the whole provided block](https://github.com/wordpress-mobile/WordPress-Android/blob/860a5a723842fe3d178aec639735b527538a1838/WordPress/src/test/java/org/wordpress/android/CoroutinesUtils.kt#L18) in "runBlocking" so that should be enough, right?

Reply:
Replaced in 6f295d1